go/vt/vtadmin/api.go

```diff
@@ -242,6 +243,154 @@ func (api *API) GetKeyspaces(ctx context.Context, req *vtadminpb.GetKeyspacesReq
 	}, nil
 }
 
+// GetSchemas is part of the vtadminpb.VTAdminServer interface.
+func (api *API) GetSchemas(ctx context.Context, req *vtadminpb.GetSchemasRequest) (*vtadminpb.GetSchemasResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.GetSchemas")
+	defer span.Finish()
+
+	clusters, _ := api.getClustersForRequest(req.ClusterIds)
+
+	var (
+		schemas []*vtadminpb.Schema
+		wg      sync.WaitGroup
+		er      concurrency.AllErrorRecorder
+		m       sync.Mutex
+	)
+
+	for _, c := range clusters {
+		wg.Add(1)
+
+		// Get schemas for the cluster
+		go func(c *cluster.Cluster) {
+			defer wg.Done()
+
+			// Since tablets are per-cluster, we can fetch them once
+			// and use them throughout the other waitgroups.
+			tablets, err := api.getTablets(ctx, c)
+			if err != nil {
+				er.RecordError(err)
+				return
+			}
+
+			ss, err := api.getSchemas(ctx, c, tablets)
+			if err != nil {
+				er.RecordError(err)
+				return
+			}
+
+			m.Lock()
+			schemas = append(schemas, ss...)
+			m.Unlock()
+		}(c)
+	}
+
+	wg.Wait()
+
+	if er.HasErrors() {
+		return nil, er.Error()
+	}
+
+	return &vtadminpb.GetSchemasResponse{
+		Schemas: schemas,
+	}, nil
+}
+
+// getSchemas returns all of the schemas across all keyspaces in the given cluster.
+func (api *API) getSchemas(ctx context.Context, c *cluster.Cluster, tablets []*vtadminpb.Tablet) ([]*vtadminpb.Schema, error) {
```

Comment:
TIOLI: I know we talked a bit about this file potentially growing very large as the API surface area grows, and these methods don't actually depend on `api` for anything, so if you wanted to you could make these top-level private functions in another file, but also this is fine as-is.

Reply:
I'll leave it for now. I agree with you on one file offering better readability. Even though these methods don't use `api` directly, the colocation seems good. One day I'll get used to the verbosity.